### PR TITLE
AD-1117: Add region / create global cluster

### PR DESCRIPTION
**Pull request by mvcninja**
- Refactored `CreateClusterWizard` into separate reusable classes
- Fixed bug in wizard when using nested async wizard
- Add `CreateGlobalClusterWizard` class / unit test
- Register addRegion command
- Unit tests

### CODE REVIEW

File: packages/core/src/docdb/commands/addRegion.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import { ToolkitError } from '../../shared'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { isValidResponse } from '../../shared/wizards/wizard'
+import { CancellationError } from '../../shared/utilities/timeoutUtils'
+import { CreateGlobalClusterWizard } from '../wizards/createGlobalClusterWizard'
+import { CreateDBClusterMessage } from '@aws-sdk/client-docdb'
+import { createInstancesForCluster } from './createCluster'
+
+export async function addRegion(node: DBClusterNode): Promise<void> {
+    if (!node) {
```
Comment (kmcginnes) [inline packages/core/src/docdb/commands/addRegion.ts:21]: What is this checking for? The `node: DBClusterNode` parameter does not seem to be nullable. Can the value be falsy?

Comment (mvcninja) [reply packages/core/src/docdb/commands/addRegion.ts:21]: Commands can be invoked indirectly anywhere in vscode, so this is just a safety check.

File: packages/core/src/docdb/commands/createCluster.ts [hunk 2]
```diff
@@ -28,15 +29,15 @@ export async function createCluster(node?: DocumentDBNode) {
...
+        const wizard = new CreateClusterWizard(node?.client, {})
         const result = await wizard.run()
 
         if (!result) {
             getLogger().info('CreateCluster canceled')
             throw new ToolkitError('User cancelled wizard', { cancelled: true })
         }
 
-        const clusterName = result.ClusterName
+        const clusterName = result.RegionalCluster.DBClusterIdentifier ?? result.ElasticCluster.clusterName
```
Comment (kmcginnes) [inline packages/core/src/docdb/commands/createCluster.ts:40]: Nit: I think I've seen this logic a couple of times. Should it be encapsulated as a computed property?

Comment (kmcginnes) [inline packages/core/src/docdb/commands/createCluster.ts:40]: Nit: I think I've seen this logic a couple of times. Should it be encapsulated as a computed property?

**Review by kmcginnes — APPROVED** [review]

File: packages/core/src/docdb/commands/createCluster.ts (hunk 2 above)
Comment (mvcninja) [reply packages/core/src/docdb/commands/createCluster.ts:40]: possibly

**Pull request merged**